poisson2d/modern.f90

```diff
@@ -0,0 +1,83 @@
+module rho_interface
+  !! Define a scalar function over 2-space.
+  implicit none
+
+  private
+  public :: rho, dp
+
+  integer, parameter :: precision = 15, range = 307
+  integer, parameter :: dp = selected_real_kind(precision, range)
+
+  interface 
+    pure real(dp) module function rho(x,y)
+      !! Poisson equation inhomogeneous term.
+      implicit none
+      real(dp), intent(in) :: x,y
+    end function
+  end interface
+
+end module 
+
+submodule(rho_interface) rho_definition
+  implicit none
+contains
+  module procedure rho
+      !! To Do: give meaningful names to the magic numbers.
+      !! See https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants.
+      if (all([x,y]>0.6_dp .and. [x,y]<0.8_dp)) then
+        rho = 1.
```

Comment:
I think this has to be either:
```suggestion
        rho = 1._dp
```
or
```suggestion
        rho = 1
```
I usually prefer the latter.

Reply:
@certik GitHub shows these changes as "outdated" -- presumably because some other commits happened first or something.  I'll see whether unresolving the comment facilitates commtting the suggestion in the browser.  If not, I'll make the second edit locally and push it.

Reply:
Nope.  Unresolving didn't work.